tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -133,6 +134,111 @@ public TypeMetadataHelper()
             }
         }
 
+        public void CompareMethodMetadata(
+            CmdletMetadata cmdlet,
+            TypeMetadata oldType,
+            TypeMetadata newType,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            var oldMethods = oldType.Methods;
+            var newMethods = newType.Methods;
+
+            foreach (var oldMethod in oldMethods)
+            {
+                bool found = false;
+                var candidateMethods = newMethods.Where(m => string.Equals(m.Name, oldMethod.Name)).ToList();
```

Comment:
Are we considering method return type as well?

Reply:
Added `ReturnType` property to `MethodMetadata` and am now checking for breaking changes in the return types when a match (name + parameters) is found.